src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(last, value,null);
+        if (size == 0) {
+            first = node;
+        } else {
+            last.next = node;
+        }
+        last = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            Node<T> currentNode = new Node<>(null, value, first);
+            first.prev = currentNode;
+            first = currentNode;
+        } else {
+            Node<T> oldNode = findNode(index);
+            Node<T> currentNode = new Node<>(oldNode.prev, value, oldNode);
+            oldNode.prev.next = currentNode;
+            oldNode.prev = currentNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> oldNode = findNode(index);
+        T oldNodeValue = oldNode.item;
+        oldNode.item = value;
+        return oldNodeValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> oldNode = findNode(index);
+        T oldNodeItem = oldNode.item;
+        unlink(oldNode);
+        return oldNodeItem;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = first;
+        if (node != null) {
+            for (int i = 0; i < size; i++) {
+                if (node.item == object || node.item.equals(object)) {
+                    unlink(node);
+                    return true;
+                }
+                node = node.next;
+            }
+        }
```

Comment:
In this case, you can`t remove a first null element. You should check in the loop.
```suggestion
            for (int i = 0; i < size; i++) {
                if (object == null ? node.item == null : object.equals(node.item)) {
                    unlink(node);
                    return true;
                }
                node = node.next;
        }
```